File: 1051. Height Checker/1051.java
// 1051. Height Checker solution in Java
// https://leetcode.com/problems/height-checker/description/

class Solution {
    public int heightChecker(int[] heights) {
        int incorrect_students = 0;
        int sorted_array[] = heights.clone();
        
        Arrays.sort(sorted_array);
        
        for (int i = 0; i < heights.length; i++) {
            if (heights[i] != sorted_array[i]) {
                incorrect_students++;
            }
        }
        
        return incorrect_students;
    }
}

File: 1051. Height Checker/1051.js
// 1051. Height Checker solution in JavaScript
// https://leetcode.com/problems/height-checker/description/

/**
 * @param {number[]} heights
 * @return {number}
 */
var heightChecker = function(heights) {
    let height_count = new Array(101).fill(0);

    for (let i = 0; i < heights.length; i++) {
        height_count[heights[i]]++;
    }

    let current_height = 0;
    let result = 0;

    for (let i = 0; i < heights.length; i++) {
        while (height_count[current_height] == 0) {
            current_height++;
        }

        if (current_height != heights[i]) {
            result++;
        }

        height_count[current_height]--;
    }

    return result;
};